VAMobile/src/store/slices/lettersSlice.ts

```diff
@@ -79,10 +79,10 @@ export const getLetterBeneficiaryData =
   async (dispatch) => {
     dispatch(dispatchClearErrors(screenID))
     dispatch(dispatchSetTryAgainFunction(() => dispatch(getLetterBeneficiaryData(screenID))))
-    dispatch(dispatchStartGetLetterBeneficiaryData())
 
     try {
       const letterBeneficiaryData = await api.get<api.LetterBeneficiaryDataPayload>('/v0/letters/beneficiary')
+      dispatch(dispatchStartGetLetterBeneficiaryData())
```

Comment:
After looking into this, it seems like the root of the issue is that both `dispatchStartGetLetterBeneficiaryData` and `dispatchStartGetLetters` (called from the previous screen) set the same `loading` state to `true`, which is causing the previous screen to load while navigating to the next screen. The reason moving the dispatch after the API call kind of works is because it gives the app a little more time to navigate away from the letters screen, so you don't see the loading component on that screen even though it's probably still being rendered.

Having a separate loading state for beneficiary data, maybe something like `loadingLetterBeneficiaryData` and updating `dispatchStartGetLetterBeneficiaryData` to set that instead, would fix the root issue. This is a great example of what React Query will help us avoid; needing to manually manage different loading state.

Also, while testing this, I think I found a new bug (unrelated to this ticket). Whenever the API call for fetching beneficiary data fails, the screen is stuck on the loading component. This is due to the [logic](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/123a0b450054423581e953e65874a962747f2958/VAMobile/src/screens/BenefitsScreen/Letters/BenefitSummaryServiceVerification/BenefitSummaryServiceVerification.tsx#L209-L215) for rendering the error component not accounting for when there's an error loading beneficiary data. I'm gonna spin up a new bug ticket for that

Reply:
updated to use this approach, can't wait to move this on over to react query lol